File: android/app/src/main/java/com/caddieai/android/data/model/ShotContext.kt
package com.caddieai.android.data.model

import kotlinx.serialization.Serializable

@Serializable
data class ShotContext(
    val distanceToPin: Int = 150,
    val shotType: ShotType = ShotType.APPROACH,
    val lie: LieType = LieType.FAIRWAY,
    val windStrength: WindStrength = WindStrength.CALM,
    val windDirection: WindDirection = WindDirection.NONE,
    val slope: Slope = Slope.FLAT,
    val aggressiveness: Aggressiveness? = null, // null = use profile default
    val elevationChangeYards: Int = 0, // positive = uphill
    val hazardNotes: String = "",
    val holeNumber: Int? = null,
    val par: Int? = null,
    val currentScore: Int? = null,
    val pinPosition: String = "", // e.g. "front left", "back right"
    val greenFirmness: String = "", // e.g. "soft", "medium", "firm"
    val playerLocation: GeoPoint? = null,
    val pinLocation: GeoPoint? = null,
)

File: ios/CaddieAI/Models/ShotContext.swift
//
//  ShotContext.swift
//  CaddieAI
//

import Foundation

struct ShotContext: Codable, Sendable {
    var distanceYards: Int
    var shotType: ShotType
    var lieType: LieType
    var windStrength: WindStrength
    var windDirection: WindDirection
    var slope: Slope
    var aggressiveness: Aggressiveness
    var hazardNotes: String

    static var `default`: ShotContext {
        ShotContext(
            distanceYards: 150,
            shotType: .approach,
            lieType: .fairway,
            windStrength: .none,
            windDirection: .into,
            slope: .level,
            aggressiveness: .normal,
            hazardNotes: ""
        )
    }
}
